src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -380,18 +412,68 @@ private Integer validateModelAndDimension(
       Integer configVectorDimension = model.vectorDimension().get();
       if (userVectorDimension == null) {
-        return configVectorDimension; // Use config dimension if user didn't provide one
+        return configVectorDimension; // Use model's dimension if user hasn't specified any
       } else if (!configVectorDimension.equals(userVectorDimension)) {
         throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
-            "The provided dimension value '%s' doesn't match the model supports dimension value '%s'",
+            "The provided dimension value '%s' doesn't match the model's supported dimension value '%s'",
             userVectorDimension, configVectorDimension);
       }
       return configVectorDimension;
     }
+
+    // Handle models with a range of acceptable dimensions
+    return model.parameters().stream()
+        .filter(param -> param.name().equals("vectorDimension"))
+        .findFirst()
+        .map(param -> validateRangeDimension(param, userVectorDimension))
+        .orElse(userVectorDimension); // should not go here
+  }
+
+  /**
+   * Validates the user-provided vector dimension against the dimension parameter's validation
+   * constraints.
+   *
+   * @param param the parameter configuration containing validation constraints
+   * @param userVectorDimension the vector dimension provided by the user
+   * @return the appropriate vector dimension based on parameter configuration
+   * @throws ApiException if the user-provided dimension is not valid
+   */
+  private Integer validateRangeDimension(
+      EmbeddingProvidersConfig.EmbeddingProviderConfig.ParameterConfig param,
+      Integer userVectorDimension) {
+    // Use the default value if the user has not provided a dimension
+    if (userVectorDimension == null) {
+      return Integer.valueOf(param.defaultValue().get());
+    }
+
+    // Extract validation type and values for comparison
+    Map.Entry<EmbeddingProvidersConfig.EmbeddingProviderConfig.ValidationType, List<Integer>>
+        entry = param.validation().entrySet().iterator().next();
+    EmbeddingProvidersConfig.EmbeddingProviderConfig.ValidationType validationType = entry.getKey();
+    List<Integer> validationValues = entry.getValue();
+
+    // Perform validation based on the validation type
+    switch (validationType) {
+      case NUMERIC_RANGE -> {
+        if (userVectorDimension < validationValues.get(0)
+            || userVectorDimension > validationValues.get(1)) {
+          throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+              "The provided dimension value '%s' is not within the supported numeric range",
+              userVectorDimension);
+        }
+      }
+      case OPTIONS -> {
+        if (!validationValues.contains(userVectorDimension)) {
+          throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+              "The provided dimension value '%s' is not within the supported options",
```

Comment:
Same here, include List of accepted values (can be added via `%s` format parameter)